pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1149,7 +1149,7 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 		}
 
 		// Fallback to file mount secret instead of SDS if meshConfig.sdsUdsPath isn't set or tls.mode is TLSSettings_MUTUAL.
-		if !node.Metadata.SdsEnabled || opts.push.Mesh.SdsUdsPath == "" || tls.Mode == networking.TLSSettings_MUTUAL {
+		if opts.push.Mesh.SdsUdsPath == "" || tls.Mode == networking.TLSSettings_MUTUAL {
```

Comment:
I am fairly sure this will break upgrades, because we will send SDS config to 1.4 proxies that do not understand them

Reply:
With this PR, during the test (go test -v -p 1 ./tests/integration/security/sds_citadel_control_plane_auth_disabled --istio.test.ci --istio.test.env kube --istio.test.kube.minikube), the ingress gateway is able to be detected as ready.  Without this PR, the log (https://storage.googleapis.com/istio-prow/logs/integ-k8s-115_istio_release-1.5_postsubmit/78/build-log.txt) shows that ingress gateway was never ready, which caused the test failure in https://github.com/istio/istio/issues/20995.

This PR reverts the cluster config changes introduced in https://github.com/istio/istio/pull/20959, which breaks ingress gateway in the sds_citadel_control_plane_auth_disabled test.

We need more context to understand the requirements for upgrade and what a proper solution for upgrade should be. As this PR is to fix the sds_citadel_control_plane_auth_disabled test, how about we open a separate issue to list the requirements for upgrade and design a proper solution for upgrade?